lib/extensions/ar_virtual.rb

```diff
@@ -143,6 +143,7 @@ def define_delegate(method_name, method, to: nil, allow_nil: nil, default: nil)
       if allow_nil
         method_def = [
           "def #{method_name}(#{definition})",
+          "return self[:#{method_name}] if has_attribute?(:#{method_name})",
```

Comment:
This code is forked from active support's `delegate`.
Wanted to keep it as similar to original code as possible.

You are right, HEREDOC would be easier to read.